File: CareConnect/CareConnect.Models/SearchObjects/ChildSearchObject.cs
using System;
using System.Collections.Generic;
using System.Text;
using CareConnect.Models.SearchObjects;

namespace CareConnect.Models.SearchObjects
{
    public class ChildSearchObject : BaseSearchObject<ChildAdditionalData>
    {
        public string? FirstNameGTE { get; set; }

        public string? LastNameGTE { get; set; }

        public DateTime? BirthDateGTE { get; set; }

        public DateTime? BirthDateLTE { get; set; }

        public String? Gender { get; set; }
    }
}

File: UI/careconnect_mobile/lib/models/search_objects/child_search_object.dart
import 'package:careconnect_mobile/models/search_objects/child_additional_data.dart';
import 'package:json_annotation/json_annotation.dart';

part 'child_search_object.g.dart';

@JsonSerializable(explicitToJson: true)
class ChildSearchObject {
  String? fts;
  String? firstNameGTE;
  String? lastNameGTE;
  DateTime? birthDateGTE;
  DateTime? birthDateLTE;
  String? gender;
  int? page;
  String? sortBy;
  bool? sortAscending;
  ChildAdditionalData? additionalData;
  bool? includeTotalCount;

  ChildSearchObject({
    this.fts,
    this.firstNameGTE,
    this.lastNameGTE,
    this.birthDateGTE,
    this.birthDateLTE,
    this.gender,
    this.page,
    this.sortBy,
    this.sortAscending,
    this.additionalData,
    this.includeTotalCount,
  });

  factory ChildSearchObject.fromJson(Map<String, dynamic> json) =>
      _$ChildSearchObjectFromJson(json);

  Map<String, dynamic> toJson() => _$ChildSearchObjectToJson(this);
}
